debugger/src/main/antlr/RustTypeName.g4

```diff
@@ -0,0 +1,176 @@
+/*
+ * Grammar rules for type names generated by rustc for the debug info.
+ *
+ * For the details about debug info type names, see:
+ * https://github.com/rust-lang/rust/blob/9da4644d5685aa0c4daa4aea6ddc9eb834ae51cc/compiler/rustc_codegen_ssa/src/debuginfo/type_names.rs
+ * https://github.com/rust-lang/rust/blob/9da4644d5685aa0c4daa4aea6ddc9eb834ae51cc/compiler/rustc_codegen_llvm/src/debuginfo/metadata/enums/cpp_like.rs
+*/
+
+grammar RustTypeName;
+
+parseTypeName : typeName EOF ;
+
+typeName
+    : msvcTypeName
+    | commonTypeName
+    ;
+
+msvcTypeName
+    : msvcStr
+    | msvcNever
+    | msvcTuple
+    | msvcPtr_const
+    | msvcPtr_mut
+    | msvcRef
+    | msvcRef_mut
+    | msvcArray
+    | msvcSlice
+    | msvcEnum
+    | msvcEnum2
+    ;
+
+msvcStr: STR ;
+
+msvcNever: NEVER ;
+
+msvcTuple
+    : TUPLE LT items=commaSeparatedList? GT
+    ;
+
+msvcPtr_const
+    : PTR_CONST LT type=typeName GT
+    ;
+
+msvcPtr_mut
+    : PTR_MUT LT type=typeName GT
+    ;
+
+msvcRef
+    : REF LT type=typeName GT
+    ;
+
+msvcRef_mut
+    : REF_MUT LT type=typeName GT
+    ;
+
+msvcArray
+    : ARRAY LT type=typeName COMMA length=WORD GT
+    ;
+
+msvcSlice
```

Comment:
Do I understand correctly that msvc types don't contain info about mutability of arrays and slices?

Reply:
Yes, it seems so as they are not wrapped into `ref$<_>`/`ref_mut$<_>` for some reason